### PR TITLE
Bump rekor, fulcio, sigstore deps to avoid pulling in vulnerable go-tuf version.

### PR DESCRIPTION
**Pull request by dhaus67**
#### Summary
`go-tuf` has a vulnerability assigned: [GHSA-66x3-6cw3-v5gj](https://github.com/theupdateframework/go-tuf/security/advisories/GHSA-66x3-6cw3-v5gj).

The versions of `github.com/sigstore/fulcio`, `github.com/sigstore/rekor`, and `github.com/sigstore/sigstore` still pull in the vulnerable version. 
By bumping those versions, the vulnerable versions will not be pulled in anymore.

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by haydentherapper** [comment]
Thanks for the change! Some test failures. 
FWIW, only Cosign actually uses go-tuf directly and was affected.

**Comment by dhaus67** [comment]
@haydentherapper thanks for the quick update! I've fixed the tests + run the deps update.

File: go.mod [hunk 1]
```diff
@@ -55,9 +55,9 @@ require (
 	github.com/pierrec/lz4 v2.6.1+incompatible
 	github.com/ryanuber/go-glob v1.0.0
 	github.com/secure-systems-lab/go-securesystemslib v0.3.1
-	github.com/sigstore/fulcio v0.1.2-0.20220114150912-86a2036f9bc7
-	github.com/sigstore/rekor v0.4.1-0.20220114213500-23f583409af3
-	github.com/sigstore/sigstore v1.2.1-0.20220424143412-3d41663116d5
+	github.com/sigstore/fulcio v0.4.2-0.20220531170627-fa41bd927942
+	github.com/sigstore/rekor v0.7.1-0.20220531123351-0c1de2a6239c
```
Comment (janisz) [inline go.mod:59]:
```suggestion
	github.com/sigstore/rekor v0.10.0
```

**Comment by dlorenc** [comment]
I think this one can be closed now.

**Pull request closed without merging**